### PR TITLE
Graph partition

### LINKED ISSUE #2338: Graph partition

### PR DESCRIPTION
**Pull request by 0x00b1**
Resolves #2338.

### CODE REVIEW

File: cellprofiler/modules/graphpartition.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
+"""
+
+Graph partition
```
Comment (mcquin) [inline cellprofiler/modules/graphpartition.py]: Documentation

File: cellprofiler/modules/graphpartition.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import cellprofiler.image
+import cellprofiler.module
+import cellprofiler.object
+import cellprofiler.setting
+import skimage.measure
+import skimage.segmentation
+
+
+class GraphPartition(cellprofiler.module.Module):
+    category = "Image segmentation"
```
Comment (mcquin) [inline cellprofiler/modules/graphpartition.py]: We need consistency between title case and not title case for categories.

File: cellprofiler/modules/graphpartition.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    module_name = "Graph partition"
```
Comment (mcquin) [inline cellprofiler/modules/graphpartition.py]: We need consistency between camel case and whatever this format is called (regular?).

File: cellprofiler/modules/graphpartition.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    variable_revision_number = 1
+
+    def create_settings(self):
+        self.image_name = cellprofiler.setting.ImageNameSubscriber(
+            "Image"
+        )
+
+        self.object_name = cellprofiler.setting.ObjectNameProvider(
+            "Object",
+            ""
```
Comment (mcquin) [inline cellprofiler/modules/graphpartition.py]: Is there any sane default for object names?

File: cellprofiler/modules/graphpartition.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        markers_data = markers.pixel_data
+
+        label_data = skimage.segmentation.random_walker(
+            data=image_data,
+            labels=markers_data,
+            beta=130,
+            mode="bf",
+            tol=0.001,
+            copy=True,
+            multichannel=False,
```
Comment (mcquin) [inline cellprofiler/modules/graphpartition.py]: `multichannel` should be set to `True` for multi-dimensional 2D images.

File: cellprofiler/modules/graphpartition.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+            return_full_prob=False,
+            spacing=None
```
Comment (mcquin) [inline cellprofiler/modules/graphpartition.py]: `spacing` should be set for 3D images.

File: cellprofiler/modules/graphpartition.py [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        )
+
+        label_data = skimage.measure.label(label_data)
+
+        objects = cellprofiler.object.Objects()
+
+        objects.segmented = label_data
```
Comment (mcquin) [inline cellprofiler/modules/graphpartition.py]: 3D segmentations are not yet supported on master.

File: cellprofiler/modules/graphpartition.py [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+            beta=130,
+            mode="bf",
+            tol=0.001,
+            copy=True,
+            multichannel=False,
+            return_full_prob=False,
+            spacing=None
+        )
+
+        label_data = skimage.measure.label(label_data)
```
Comment (mcquin) [inline cellprofiler/modules/graphpartition.py]: `skimage.measure.label` returns `labels` and `num`. Only `labels` are needed here.

File: cellprofiler/modules/graphpartition.py [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+            workspace.display_data.label_data = label_data
+
+            workspace.display_data.dimensions = dimensions
+
+    def display(self, workspace, figure):
+        figure.set_subplots((1, 2))
+
+        figure.subplot_imshow(0, 0, workspace.display_data.image_data, dimensions=workspace.display_data.dimensions)
+
+        figure.subplot_imshow(0, 1, workspace.display_data.label_data, dimensions=workspace.display_data.dimensions)
```
Comment (mcquin) [inline cellprofiler/modules/graphpartition.py]: Would it be helpful to overlay the labels on the image?

Comment (0x00b1) [inline cellprofiler/modules/graphpartition.py]: @AnneCarpenter The CellProfiler term is “segmentation” (i.e. `cellprofiler.object.Segmentation`).

Comment (AnneCarpenter) [inline cellprofiler/modules/graphpartition.py]: Ohhh. That kind of labels. thanks!

Comment (0x00b1) [inline cellprofiler/modules/graphpartition.py]: @AnneCarpenter No problem. 😎

**Comment by 0x00b1** [comment]
Replaced by #2391!

**Pull request closed without merging**